test/system/src/test/java/io/pravega/test/system/PravegaTest.java

```diff
@@ -155,7 +155,7 @@ public void simpleTest() {
                 fail("Reinitialization Exception is not expected");
             }
             // try reading until all the written events are read, else the test will timeout.
-        } while ((event.getEvent() != null || event.isCheckpoint()) && readCount < NUM_EVENTS);
+        } while ((event.getEvent() == null || event.isCheckpoint()) && readCount < NUM_EVENTS);
```

Comment:
With this change, if the event is a regular read (contains data and is not a checkpoint), the loop will exit, right?

Looking at the logs, the problem I see is the following:

- The writer seems to correctly write events to the Stream (segment store logs): 
```
2018-12-17 09:13:57,493 12934340 [epollEventLoopGroup-3-2] DEBUG i.p.s.s.h.h.ServerConnectionInboundHandler - Processing request: WireCommands.ReadSegment(type=READ_SEGMENT, segment=testScopeSampleYndrFi/testStreamSampleY/0.#epoch.0, offset=0, suggestedLength=1048576, delegationToken=)
2018-12-17 09:13:57,493 12934340 [epollEventLoopGroup-3-2] DEBUG i.p.s.s.c.StreamSegmentContainer - SegmentContainer[3]: read [testScopeSampleYndrFi/testStreamSampleY/0.#epoch.0, 0, 1048576]
2018-12-17 09:13:57,493 12934340 [epollEventLoopGroup-3-2] DEBUG i.p.s.s.reading.ContainerReadIndex - ReadIndex[3]: read (StreamSegmentId = 12402, Offset = 0, MaxLength = 1048576).
2018-12-17 09:13:57,493 12934340 [epollEventLoopGroup-3-2] DEBUG i.p.s.s.r.StreamSegmentReadIndex - ReadIndex[3-12402]: Read (Offset = 0, MaxLength = 1048576).
2018-12-17 09:13:57,496 12934343 [epollEventLoopGroup-2-1] INFO  i.n.handler.logging.LoggingHandler - [id: 0xa441dab2, L:/xxx:12345] READ: [id: 0x026f5ed8, L:/xxx:12345 - R:/xxx:56880]
2018-12-17 09:13:57,496 12934343 [epollEventLoopGroup-2-1] INFO  i.n.handler.logging.LoggingHandler - [id: 0xa441dab2, L:/xxx:12345] READ COMPLETE
```

- The client issues the `READ_SEGMENT` request:
```
2018-12-17 09:13:57,491 3485 [grpc-default-executor-0] TRACE i.p.c.s.i.AsyncSegmentInputStreamImpl - Sending read request WireCommands.ReadSegment(type=READ_SEGMENT, segment=testScopeSampleYndrFi/testStreamSampleY/0.#epoch.0, offset=0, suggestedLength=1048576, delegationToken=)
```

- The segment store receives that request:
```
2018-12-17 09:13:57,493 12934340 [epollEventLoopGroup-3-2] DEBUG i.p.s.s.h.h.ServerConnectionInboundHandler - Processing request: WireCommands.ReadSegment(type=READ_SEGMENT, segment=testScopeSampleYndrFi/testStreamSampleY/0.#epoch.0, offset=0, suggestedLength=1048576, delegationToken=)
2018-12-17 09:13:57,493 12934340 [epollEventLoopGroup-3-2] DEBUG i.p.s.s.c.StreamSegmentContainer - SegmentContainer[3]: read [testScopeSampleYndrFi/testStreamSampleY/0.#epoch.0, 0, 1048576]
2018-12-17 09:13:57,493 12934340 [epollEventLoopGroup-3-2] DEBUG i.p.s.s.reading.ContainerReadIndex - ReadIndex[3]: read (StreamSegmentId = 12402, Offset = 0, MaxLength = 1048576).
2018-12-17 09:13:57,493 12934340 [epollEventLoopGroup-3-2] DEBUG i.p.s.s.r.StreamSegmentReadIndex - ReadIndex[3-12402]: Read (Offset = 0, MaxLength = 1048576).
2018-12-17 09:13:57,496 12934343 [epollEventLoopGroup-2-1] INFO  i.n.handler.logging.LoggingHandler - [id: 0xa441dab2, L:/xxx:12345] READ: [id: 0x026f5ed8, L:/10.200.91.79:12345 - R:/xxx:56880]
2018-12-17 09:13:57,496 12934343 [epollEventLoopGroup-2-1] INFO  i.n.handler.logging.LoggingHandler - [id: 0xa441dab2, L:/xxx:12345] READ COMPLETE
```

- But the client is not able to read any data:
```
2018-12-17 09:13:58,497 4491 [Time-limited test] DEBUG io.pravega.test.system.PravegaTest - Read event: null.
```

Perhaps, a possible cause for this is that the timeout may be too low in this infrastructure (it was not the case before, though).

Reply:
In a second execution, it looks like a single event was read, which may indicate that read can happen, but the timeout is too short:
```
Caused by: java.lang.AssertionError: Read count should be equal to write count expected:<100> but was:<1>
```

Reply:
the test tries to read until it has read the required number of events. Else the test will timeout.